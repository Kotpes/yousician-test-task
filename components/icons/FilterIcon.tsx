interface Props {
    onClick: () => void;
    fill: string;
    className?: string;
}
const FilterIcon = ({ fill = '#fff', onClick, className }: Props) => {
    return (
        <div onClick={onClick} role="button" tabIndex={0} onKeyPress={onClick} className={className}>
            <svg width={16} height={16} viewBox="0 0 16 16">
                <path d="M0 0h16v16H0z" fill="none" />
                <path fill={fill} d="M12.5 3.104a2.751 2.751 0 010 5.292v6.854a.75.75 0 11-1.5 0V8.396a2.751 2.751 0 010-5.292V.75a.75.75 0 111.5 0v2.354zM5 7.604a2.751 2.751 0 010 5.292v2.354a.75.75 0 11-1.5 0v-2.354a2.751 2.751 0 010-5.292V.75a.75.75 0 011.5 0v6.854zM4.25 11.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5zm7.5-4.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5z" />
            </svg>
        </div>

    )
}

export default FilterIcon