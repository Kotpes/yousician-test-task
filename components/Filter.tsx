import { useState, useRef } from 'react'
import range from 'lodash.range'

import FilterIcon from './icons/FilterIcon'
import styles from './Filter.module.css'

const initialValues = { firstValue: 0, secondValue: 0 }
interface Props {
    // eslint-disable-next-line no-unused-vars
    onRangeSelect: (selectedRange: number[]) => void;
}

const Filter = ({ onRangeSelect }: Props) => {
    const [selectedRange, setSelectedRange] = useState([])
    const [filtersVisible, setFiltersVisibility] = useState(false)
    const [selectedValuesLabel, setSelectedValuesLabel] = useState('')
    const values = useRef(initialValues)
    const labelText = filtersVisible ? 'Hide filter' : 'Filter by level'
    const selectedFiltersClass = selectedValuesLabel ? styles.filtersSelected : ''
    const filterIconFill = filtersVisible ? '#fff' : '#000'
    const filterIconBackground = filtersVisible ? '' : styles.active

    const toggleFilters = () => {
        setFiltersVisibility(!filtersVisible)
    }

    const handleSelectedRange = () => {
        const { firstValue, secondValue } = values.current
        const edges = firstValue > secondValue ? { start: secondValue, end: firstValue } : { start: firstValue, end: secondValue }
        const selectedRange = range(edges.start, edges.end + 1)
        const selectedValueLabel = `${selectedRange[0]} - ${selectedRange[selectedRange.length - 1]}`

        setSelectedRange(selectedRange)
        onRangeSelect(selectedRange)
        setSelectedValuesLabel(selectedValueLabel)
    }

    const handleFilterSelection = (value: number) => {
        const { firstValue, secondValue } = values.current

        if (firstValue > 0 && secondValue > 0) {
            setSelectedRange([value])
            values.current.firstValue = value
            values.current.secondValue = 0
        } else if (!firstValue) {
            const selected = [...selectedRange, value]
            setSelectedRange(selected)
            values.current.firstValue = value
        } else if (!secondValue) {
            values.current.secondValue = value
            handleSelectedRange()
        }
    }


    return (
        <section className={styles.filterContainer}>
            <span className={styles.filterLabel}>{labelText}</span>
            <div className={`${styles.selectedFilters} ${selectedFiltersClass}`}>
                {selectedValuesLabel &&
                    <span className={styles.selectedValues}>{selectedValuesLabel}</span>
                }
                <FilterIcon fill={filterIconFill} onClick={toggleFilters} className={`${styles.filterButton} ${filterIconBackground}`} />
            </div>
            {filtersVisible &&
                <section className={styles.filters}>
                    {Array.from({ length: 15 }, (v, k) => {
                        const value = k + 1;
                        const isInRange = selectedRange.includes(value)
                        const backgroundStyle = isInRange ? styles.selected : ''
                        return (
                            <span
                                className={`${styles.filter} ${backgroundStyle}`}
                                onClick={() => handleFilterSelection(value)}
                                onKeyPress={() => handleFilterSelection(value)}
                                role="button"
                                tabIndex={0}
                                key={k}
                            >
                                {value}
                            </span>
                        )
                    })}
                </section>
            }

        </section>
    )
}

export default Filter