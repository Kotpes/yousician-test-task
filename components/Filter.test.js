import { create, act } from 'react-test-renderer';

import Filter from './Filter';
import FilterIcon from './icons/FilterIcon';

const onRangeSelect = jest.fn();
const props = {
  onRangeSelect,
};

describe('Search', () => {
  test('it matches snapshots', () => {
    const component = create(<Filter {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('filter label changes on toggle', () => {
    let component;
    act(() => {
      component = create(<Filter {...props} />);
    });
    const instance = component.root;
    const filterIconComponent = instance.findByType(FilterIcon);
    const label = instance.findByProps({ id: 'label' });

    expect(label.children).toEqual(['Filter by level']);

    act(() => filterIconComponent.props.onClick());

    expect(label.children).toEqual(['Hide filter']);
  });
  test('filters visibiity is affected by toggle', () => {
    let component;
    act(() => {
      component = create(<Filter {...props} />);
    });
    const instance = component.root;
    const filterIconComponent = instance.findByType(FilterIcon);

    act(() => filterIconComponent.props.onClick());
    const filtersContainer = instance.findByProps({ className: 'filters' });
    expect(filtersContainer.children.length).toBe(15);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
