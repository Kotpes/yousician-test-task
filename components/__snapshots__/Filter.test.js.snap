// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search filters visibiity is affected by toggle 1`] = `
<section
  className="filterContainer"
>
  <span
    className="filterLabel"
    id="label"
  >
    Hide filter
  </span>
  <div
    className="selectedFilters "
  >
    
    <div
      className="filterButton "
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M0 0h16v16H0z"
          fill="none"
        />
        <path
          d="M12.5 3.104a2.751 2.751 0 010 5.292v6.854a.75.75 0 11-1.5 0V8.396a2.751 2.751 0 010-5.292V.75a.75.75 0 111.5 0v2.354zM5 7.604a2.751 2.751 0 010 5.292v2.354a.75.75 0 11-1.5 0v-2.354a2.751 2.751 0 010-5.292V.75a.75.75 0 011.5 0v6.854zM4.25 11.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5zm7.5-4.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5z"
          fill="#fff"
        />
      </svg>
    </div>
  </div>
  <section
    className="filters"
  >
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#6fc13e"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={87.9645943005142}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          1
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#6fc13e"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={81.68140899333461}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          2
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#6fc13e"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={75.39822368615503}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          3
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#6fc13e"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={69.11503837897544}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          4
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#6fc13e"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={62.83185307179584}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          5
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#ff8e00"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={56.54866776461627}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          6
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#ff8e00"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={50.26548245743668}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          7
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#ff8e00"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={43.9822971502571}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          8
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#ff8e00"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={37.69911184307752}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          9
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#ff8e00"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={31.41592653589792}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          10
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#dc001c"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={25.13274122871835}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          11
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#dc001c"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={18.84955592153876}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          12
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#dc001c"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={12.566370614359165}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          13
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#dc001c"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={6.28318530717959}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          14
        </text>
      </svg>
    </span>
    <span
      className="filter"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={45}
        viewBox="0 0 42 42"
        width={45}
      >
        <circle
          className="ring"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#383635"
          strokeDasharray={0}
          strokeWidth={1.8}
        />
        <circle
          className="progress"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#dc001c"
          strokeDasharray={94.24777960769379}
          strokeDashoffset={0}
          strokeWidth={1.8}
          transform="rotate(-90 21 21)"
        />
        <circle
          className="devider"
          cx={21}
          cy={21}
          fill="transparent"
          r={15}
          stroke="#000"
          strokeDasharray="9% 65.5%"
          strokeWidth={2}
          transform="rotate(-96 21 21)"
        />
        <text
          dy=".3em"
          fill="#fff"
          fontSize="14"
          fontWeight="800"
          textAnchor="middle"
          x="50%"
          y="50%"
        >
          15
        </text>
      </svg>
    </span>
  </section>
</section>
`;

exports[`Search it matches snapshots 1`] = `
<section
  className="filterContainer"
>
  <span
    className="filterLabel"
    id="label"
  >
    Filter by level
  </span>
  <div
    className="selectedFilters "
  >
    
    <div
      className="filterButton active"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <svg
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <path
          d="M0 0h16v16H0z"
          fill="none"
        />
        <path
          d="M12.5 3.104a2.751 2.751 0 010 5.292v6.854a.75.75 0 11-1.5 0V8.396a2.751 2.751 0 010-5.292V.75a.75.75 0 111.5 0v2.354zM5 7.604a2.751 2.751 0 010 5.292v2.354a.75.75 0 11-1.5 0v-2.354a2.751 2.751 0 010-5.292V.75a.75.75 0 011.5 0v6.854zM4.25 11.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5zm7.5-4.5a1.25 1.25 0 100-2.5 1.25 1.25 0 000 2.5z"
          fill="#000"
        />
      </svg>
    </div>
  </div>
</section>
`;
